### PR TITLE
Prepare release of v3.0.0-alpha-2

### PR DESCRIPTION
**Pull request by quantum9Innovation**
# Releasing v3.0.0-alpha-2
___

## New Feature
This PR focuses primarily on version bumps and file updates. The major changes are of course the bump to v3.0.0-alpha-2, preparing for the next alpha release, and the new release notes in the CHANGELOG. The other main feature is a new sanity check and more global variables to access data like version strings, documentation URLs, and more. The sanity check now is not run automatically, resulting in a small performance improvement.
## Known Issues
There are no known issues at this time (other than spellcheck, of course—this will be fixed soon).
## Code Breakdown
 - deepsource.toml
    - Upgrade analyzer runtime to a supported Python version
 - CHANGELOG.md
    - Add release notes for v3.0.0-alpha-2
 - README-nightly.md
    - Update from documentation
 - SECURITY.md
    - Add alpha releases to supported versions
    - Update codefactor.io to DeepSource
 - epispot/\_\_init\_\_.py
    - Add new sanity check
    - Add more global variables
    - Bring back support for `__version__`
 - epispot/fitters.py
    - Deprecate
 - setup-nightly.py
    - Bump version
    - Update requirements specs
 - setup.cfg
    - Update supported Python versions
 - tests/test_x.py
    - Test sanity checks manually
## Additional Notes
As this PR changes a lot of manifest files, @Quantalabs is going to lead the review process as code owner. We will need to merge this fast and then rebase the new changes onto the v3.0.0-alpha branch before deploying.

A full comparison with v3.0.0-alpha-1 can be found here:
https://github.com/epispot/epispot/compare/v3.0.0-alpha-1...v3a2

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -14,6 +14,11 @@ exclude_patterns = [
 [[analyzers]]
 name = "python"
 enabled = true
+dependency_file_paths = [
+    "requirements/requirements.txt",
+    "requirements/requirements-nightly.txt",
+    "requirements/requirements-dev.txt"
+]
```
Comment (Quantalabs) [inline .deepsource.toml:21]: Would this work?

```suggestion
dependency_file_paths = [
    "requirements/*"
]
```

File: README-nightly.md [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 A Python package for the mathematical modeling of infectious diseases via compartmental models. Originally designed for epidemiologists, epispot can be adapted for almost any type of modeling scenario.
 
-> This is a nightly version of epispot and may contain possibly unstable code.\
-> **Please see usage instructions prior to adding this project as a dependency**\
+> This is a nightly version of epispot and may contain possibly unstable code.
```
Comment (Quantalabs) [inline README-nightly.md:9]:
```suggestion
> This is a alpha version of epispot and may contain possibly unstable code.  
```

File: setup.cfg [hunk 3]
```diff
@@ -23,7 +23,7 @@ classifiers =
 package_dir =
     = epispot
 packages = find:
-python_requires = >=3.6
+python_requires = >=3.7
```
Comment (Quantalabs) [inline setup.cfg:26]: Why deprecate the support for 3.6?

**Review by Quantalabs — CHANGES_REQUESTED** [review]

File: .deepsource.toml (hunk 1 above)
Comment (quantum9Innovation) [reply .deepsource.toml:21]: No, `requirements/README.md` is not a requirements file and `requirements/pre-requirements.txt` does not actually contain any useful dependency information.

File: README-nightly.md (hunk 2 above)
Comment (quantum9Innovation) [reply README-nightly.md:9]: While that's technically correct, it's a headache to maintain and not a meaningful change. There are some differences between nightly and alpha releases, but those differences are mentioned when appropriate. Here, differentiating between the two release types is unnecessary.

File: setup.cfg (hunk 3 above)
Comment (quantum9Innovation) [reply setup.cfg:26]: Python 3.6 is no longer included in our testing suite as a result of #93. Additionally, this change was already made to `setup-nightly.py` a long time ago. This is really a minor change and has already been detailed in the CHANGELOG.

**Review by Quantalabs — APPROVED** [review]

**Pull request merged**